### ISSUE TITLE
Sequence not ordering actions properly

### ISSUE THREAD

**Issue by wilczarzgit** [Demigiant/dotween#231]
Hi there,

I have been trying to use a sequence to execute three actions one after another:

1. tween
2. block of code
3. tween

So I have tried this code:

```
		var seq = DOTween.Sequence();
		seq.Append(transform.DOMove(Vector3.one, 0.5f));
		seq.AppendCallback(() =>
		{
			DoSomething();
			transform.position = -Vector3.one;
		});
		seq.Append(transform.DOMove(Vector3.zero, 0.5f));

```
But it doesn't seem to work. Setting the transform inside the callback has no effect, almost as if the first tween is not yet finished? Now if I change the code this way, it works:

```
		transform.DOMove(Vector3.one, 0.5f).OnComplete(() =>
		{
			DOSomething();
			transform.position = -Vector3.one;
			transform.DOMove(Vector3.zero, 0.5f);
		});
```

It may be shorter but it's much harder to read, as the elements in the sequence are not clearly visible.

Somewhere online I found a tip to insert intervals between the steps. Is seems very very ugly. The sequence should take care of the order of execution (that's what sequences are for :)

Is this a bug of DOTween?

**Comment by wilczarzgit** [comment]
Anyone there? :)

**Comment by Demigiant** [comment]
Ahoy! Sorry for the delay, I'm stuck with a big upgrade to DOTween and am kind of distracted towards the rest :P

The problem there is that things that happen at the same exact time in a Sequence (when adding elements) don't have a specific order for many reasons. I never thought this could be a bug, but the way you're using it (where two things happening at the same time require a correct order) makes me think otherwise.

I'll get on this as soon as I finish the modular upgrade.

**Comment by Demigiant** [comment]
Fixed! :) Can you test this new version ([v1.2.070](http://dotween.demigiant.com/tmp/DOTween_1_2_070.zip)) out and let me know?

P.S. Please note that this update also includes the Modules upgrade, so please [follow these instructions](http://dotween.demigiant.com/support.php#before-modules) carefully when upgrading because it's a major one (and if you're using DOTween Pro instead let me know and don't test this, because you'll first need the Pro upgrade).
